lib/DiscoveryStreamFeed.jsm

```diff
@@ -32,12 +33,25 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this._prefCache = {};
   }
 
+  finalLayoutEndpoint(url, apiKey) {
+    if (url.includes("$apiKey") && !apiKey) {
+      throw new Error(`Layout Endpoint - An API key was specified but none configured: ${url}`);
+    }
+    return url.replace("$apiKey", apiKey);
+  }
+
   get config() {
     if (this._prefCache.config) {
       return this._prefCache.config;
     }
     try {
       this._prefCache.config = JSON.parse(this.store.getState().Prefs.values[PREF_CONFIG]);
+      const layoutUrl = this._prefCache.config.layout_endpoint;
+      const apiKeyPref = this._prefCache.config.api_key_pref;
+      if (layoutUrl && apiKeyPref) {
+        const apiKey = Services.prefs.getCharPref(apiKeyPref);
```

Comment:
Can you pass a second param like `""` to this just in case the pref is not defined? I think we don't want to throw in this case.

Reply:
If we are not not throwing error for empty apiKey , is it ok to return layout_endpoint url with consumer_key as ""?

Reply:
Updated with second optional param as "" to avoid throwing an exception while trying to retrieve pref

Reply:
If the url contains `$apiKey` your `finalLayoutEndpoint` function wills till throw if the result is `""`, I think that will be a more descriptive error than just the pref not existing